secret-generation.md

```diff
@@ -0,0 +1,214 @@
+# Secret Generator Extension for Service Binding
```

Comment:
it needs to be clear that this content is not a required part of the spec

Reply:
We can add a note. I need some help with phrasing it. Here is my attempt:

>Note: This spec is dependent on "Service Binding Specification for Kubernetes", but there is no direct association -- neither containment nor aggregation.

Reply:
One more sentence pointing to the release cycle:

> Note: This spec is dependent on "Service Binding Specification for Kubernetes", but there is no direct association -- neither containment nor aggregation. Both specs are going to have their independent release cycles.

Reply:
What if we moved it into a directory named "extensions" and added a README.md that contained [this paragraph](https://github.com/k8s-service-bindings/spec/pull/161/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5L630)?

Reply:
By and large, this is the issue with keeping placeholder content in the repo. It becomes murky as to what it means and why it exists.

Reply:
- Created an "extensions" directory and moved the Secret Generation extension inside
- Created "extensions/README.md" with a brief introduction and a table of extensions (No., Title, and Status columns)
- Updated the extension with a brief introduction and status section